packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -2656,6 +2656,160 @@ void main() {
       ),
     );
   });
+
```

Comment:
Good tests! Can we add an additional one which clarifies the behaviour when `backgroundColor` is present and is a `MaterialStateColor`, and `disabledBackgroundColor` is also present?